notification-server/src/bin/service.rs

```diff
@@ -346,6 +355,36 @@ async fn traverse_chain(
     processed_height
 }
 
+async fn catch_up_to_limit(
+    concordium_client: &mut Client,
+    current_height: AbsoluteBlockHeight,
+    notification_ttl: Duration,
+) -> anyhow::Result<AbsoluteBlockHeight> {
+    let current_block_height = concordium_client
+        .get_consensus_info()
+        .await?
+        .last_finalized_block_height;
+    let lower_bound = AbsoluteBlockHeight {
+        // We are not fast than 2 sec per block, hence this should be conservative enough
+        height: current_block_height.height - notification_ttl.as_secs(),
+    };
+    let lower_bound_time: chrono::DateTime<Utc> = Utc::now() - notification_ttl;
+    let time_ago_block = concordium_client
+        .find_first_finalized_block_no_earlier_than(lower_bound.., lower_bound_time)
+        .await?;
+
+    if time_ago_block.block_height > current_height {
+        info!(
+            "Skipping {} blocks",
+            time_ago_block.block_height.height - current_height.height
```

Comment:
Put the result of the calculation into a variable since the same calculation is repeated further down.